src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -112,44 +112,29 @@ public unsafe static uint[] Square(uint[] value)
                 Square(valueHigh, valueHighLength,
                        bitsHigh, bitsHighLength);
 
-                // ... compute z_a = a_1 + a_0 (call it fold...)
                 int foldLength = valueHighLength + 1;
-                uint* fold = stackalloc uint[foldLength];
-
-                Add(valueHigh, valueHighLength,
-                    valueLow, valueLowLength,
-                    fold, foldLength);
-
-                // ... compute z_1 = z_a * z_a - z_0 - z_2
                 int coreLength = foldLength + foldLength;
-                uint* core = stackalloc uint[coreLength];
-
-                Square(fold, foldLength,
-                       core, coreLength);
-                SubtractCore(bitsHigh, bitsHighLength,
-                             bitsLow, bitsLowLength,
-                             core, coreLength);
-
-                // ... and finally merge the result! :-)
-                AddSelf(bits + n, bitsLength - n, core, coreLength);
+                fixed (uint* fold = new uint[foldLength],
```

Comment:
A few days ago, @KrzysztofCwalina suggested to do both and switch depending on the size. Problem: which size? The calling code may already use the current stack near to its limit or maybe we could nearly use all of it. Deciding based on a fixed size would be easy to implement, but I've no idea how to choose that size.

We *could* just say 4096 bits (or something like that) is the limit, we'll switch to the heap then. I've really no idea if that's the way to go, but if you recommend it and name a limit, I can push a further commit accordingly.